tests/linen/module_test.py

```diff
@@ -668,6 +670,23 @@ def bar(self):
     with self.assertRaisesRegex(AttributeError, "has no attribute 'setup_called'"):
       empty.bar()
 
+  def test_module_with_attrs(self):
+    class Foo(nn.Module):
+      bar: nn.Dense = dataclasses.field(init=False)
+
+      def setup(self):
+        self.bar = nn.Dense(2)
+      
+      @nn.compact
+      def __call__(self, x):
+        return self.bar(x)
+
+    foo = Foo()
+    x = jnp.ones((2,))
+    variables = foo.init(random.PRNGKey(0), x)
+    self.assertEqual(variables['params']['bar']['kernel'].shape, (2, 2))
```

Comment:
FWIW I try to avoid expected shapes like (2, 2) because if you mixed the order it'd still be right. So I'd have `x` have shape `(3, )`. But this is a bit nit-picky

Reply:
I agree